test/integration/operator/garden/garden/garden_test.go

```diff
@@ -806,36 +806,29 @@ spec:
 		})
 
 		By("Verify that the ManagedResources related to Gardener control plane components have been deployed")
-		Eventually(func(g Gomega) []resourcesv1alpha1.ManagedResource {
-			managedResourceList := &resourcesv1alpha1.ManagedResourceList{}
-			g.Expect(testClient.List(ctx, managedResourceList, client.InNamespace(testNamespace.Name))).To(Succeed())
-			return managedResourceList.Items
-		}).Should(ContainElements(
-			MatchFields(IgnoreExtras, Fields{"ObjectMeta": MatchFields(IgnoreExtras, Fields{"Name": Equal("gardener-apiserver-runtime")})}),
-			MatchFields(IgnoreExtras, Fields{"ObjectMeta": MatchFields(IgnoreExtras, Fields{"Name": Equal("gardener-apiserver-virtual")})}),
-		))
-
-		// The garden controller waits for the Gardener-related ManagedResources to be healthy, but no
-		// gardener-resource-manager is running in this test, so let's fake this here.
-		By("Patch Gardener-related ManagedResources to report healthiness")
 		for _, name := range []string{"apiserver", "admission-controller", "controller-manager", "scheduler", "dashboard"} {
-			Eventually(makeManagedResourceHealthy("gardener-"+name+"-runtime", testNamespace.Name)).Should(Succeed())
-			Eventually(makeManagedResourceHealthy("gardener-"+name+"-virtual", testNamespace.Name)).Should(Succeed())
+			Eventually(func(g Gomega) []resourcesv1alpha1.ManagedResource {
+				managedResourceList := &resourcesv1alpha1.ManagedResourceList{}
+				g.Expect(testClient.List(ctx, managedResourceList, client.InNamespace(testNamespace.Name))).To(Succeed())
+				return managedResourceList.Items
+			}).Should(ContainElements(
+				MatchFields(IgnoreExtras, Fields{"ObjectMeta": MatchFields(IgnoreExtras, Fields{"Name": Equal("gardener-" + name + "-runtime")})}),
+				MatchFields(IgnoreExtras, Fields{"ObjectMeta": MatchFields(IgnoreExtras, Fields{"Name": Equal("gardener-" + name + "-virtual")})}),
+			), "for gardener-"+name)
+
+			// The garden controller waits for the Gardener-related ManagedResources to be healthy, but no
+			// gardener-resource-manager is running in this test, so let's fake this here.
+			By("Patch Gardener-related ManagedResources to report healthiness")
```

Comment:
Can you move the `By` out of the loop or make the step description iteration-specific?